### PR TITLE
[DOCS] Add release notes for 7.8.0

### PR DESCRIPTION
**Pull request by pugnascotia**
Add release notes for 7.8.0.

## Preview
https://elasticsearch_56340.docs-preview.app.elstc.co/diff

### CODE REVIEW

**Review by jrodewig — COMMENTED** [review]
Looks good overall. However, there are a few PRs that need to be removed.

I also left some suggestions for replacing the internal xrefs with the equivalent external link. Changing those should fix the broken docs CI test.

The remaining edits are mostly non-blocking nits.

I also tagged @lcawl  and @szabosteve on ML-related sections. I'll defer to them on those .

**Comment by lcawl** [comment]
FYI I merged the extra machine learning PRs from https://raw.githubusercontent.com/elastic/ml-cpp/7.8/docs/CHANGELOG.asciidoc

File: docs/reference/release-notes/7.8.asciidoc [hunk 1]
```diff
@@ -1,18 +1,338 @@
...
+* Fix error massage for unknown value type {pull}55821[#55821] (issue: {issue}55727[#55727])
+* Fix scripted metric in CCS {pull}54776[#54776] (issue: {issue}54758[#54758])
+* Use Decimal formatter for Numeric ValuesSourceTypes {pull}54366[#54366] (issue: {issue}54365[#54365])
+
+Allocation::
+* Fix Broken ExistingStoreRecoverySource Deserialization {pull}55657[#55657] (issue: {issue}55513[#55513])
+
+
+Authorization::
+* Resolve anonymous roles and deduplicate roles during authentication {pull}53453[#53453] (issues: {issue}31589[#31589], {issue}47195[#47195])
```
Comment (tvernum) [inline docs/reference/release-notes/7.8.asciidoc:272]: This PR was reverted. It is not in 7.8

```suggestion
```

File: docs/reference/release-notes/7.8.asciidoc [hunk 2]
```diff
@@ -173,6 +173,37 @@ Machine Learning::
 * Skip daily maintenance activity if upgrade mode is enabled {pull}54565[#54565] (issue: {issue}54326[#54326])
 * Start gathering and storing inference stats {pull}53429[#53429]
 * Unassign DFA tasks in SetUpgradeModeAction {pull}54523[#54523] (issue: {issue}54326[#54326])
```
Comment (droberts195) [inline docs/reference/release-notes/7.8.asciidoc:175]: @lcawl what is the official docs position on using the DFA acronym in public documentation? If it isn't supposed to be used then it should be expanded to "data frame analytics" here.

File: docs/reference/release-notes/7.8.asciidoc [hunk 3]
```diff
@@ -1,18 +1,338 @@
...
+* Add validation to the usage service {pull}54617[#54617]
+
+Infra/Scripting::
+* Scripting: stats per context in nodes stats {pull}54008[#54008] (issue: {issue}50152[#50152])
+
+Machine Learning::
+* Add effective max model memory limit to ML info {pull}55529[#55529] (issue: {issue}63942[#63942])
+* Add loss_function to regression {pull}56118[#56118]
+* Add new inference_config field to trained model config {pull}54421[#54421]
+* Adding failed_category_count to model_size_stats {pull}55716[#55716] (issue: {issue}1130[#1130])
```
Comment (szabosteve) [inline docs/reference/release-notes/7.8.asciidoc:164]:
```suggestion
* Add failed_category_count to model_size_stats {pull}55716[#55716] (issue: {issue}1130[#1130])
```

File: docs/reference/release-notes/7.8.asciidoc [hunk 4]
```diff
@@ -1,18 +1,338 @@
...
+* Adding prediction_field_type to inference config {pull}55128[#55128]
```
Comment (szabosteve) [inline docs/reference/release-notes/7.8.asciidoc:165]:
```suggestion
* Add prediction_field_type to inference config {pull}55128[#55128]
```

File: docs/reference/release-notes/7.8.asciidoc [hunk 5]
```diff
@@ -1,18 +1,338 @@
...
+* Improve robustness of anomaly detection to bad input data {ml-pull}1114[#1114]
+* Add new `num_matches` and `preferred_to_categories` fields to category output
+{ml-pull}1062[#1062]
+* Add mean squared logarithmic error (MSLE) for regression {ml-pull}1101[#1101]
+* Add pseudo-Huber loss for regression {ml-pull}1168[#1168]
+* Reduce peak memory usage and memory estimates for classification and regression
+{ml-pull}1125[#1125].)
+* Reduce variability of classification and regression results across our target
+operating systems {ml-pull}1127[#1127]
+* Switched data frame analytics model memory estimates from kilobytes to
```
Comment (szabosteve) [inline docs/reference/release-notes/7.8.asciidoc:193]:
```suggestion
* Switch data frame analytics model memory estimates from kilobytes to
```

**Review by szabosteve — COMMENTED** [review]
I reviewed the ML related parts and left some suggestions. Thank you!

**Comment by pugnascotia** [comment]
I've addressed the final outstanding questions. Please let me know if there is anything further to address.

**Review by jrodewig — APPROVED** [review]
LGTM. Thanks for iterating on this @pugnascotia

**Pull request merged**